tests/test_object.h

```diff
@@ -87,6 +187,119 @@ TEST_CASE("[Object] Metadata") {
 			meta_list2.size() == 0,
 			"The metadata list should contain 0 items after removing all metadata items.");
 }
+
+TEST_CASE("[Object] Construction") {
+	Object object;
+
+	CHECK_MESSAGE(
+			!object.is_reference(),
+			"Object is not a Reference.");
+
+	Object *p_db = ObjectDB::get_instance(object.get_instance_id());
+	CHECK_MESSAGE(
+			p_db == &object,
+			"The database pointer returned by the object id should reference same object.");
+}
+
+TEST_CASE("[Object] Script instance property setter") {
+	Object object;
+	_MockScriptInstance script_instance;
+	object.set_script_instance(&script_instance);
+
+	bool valid = false;
+	object.set("some_name", 100, &valid);
+	CHECK(valid);
+	Variant actual_value;
+	CHECK_MESSAGE(
+			script_instance.get("some_name", actual_value),
```

Comment:
Once the above is done:
```suggestion
			script_instance->get("some_name", actual_value),
```